File: nathan/Problema0304/problema0304.java
package Problema0304;

import java.util.Scanner;

public class problema0304 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int vCodigo[] = new int[50];
        String vNome[] = new String[50];
        int vQuant[] = new int[50];
        int vPreco[] = new int [50];   
        String auxNome;
        int i = 0, auxQuant, auxCodigo, auxPreco, nProdutos = 0;
        double somaVendas = 0;
        boolean produtoVendido = false;

        System.out.println("Digite o código do produto(-1 Para cancelar): ");
        vCodigo[i] = Integer.valueOf(input.nextLine());

        while(vCodigo[i] != -1){
            nProdutos++;
            System.out.println("Digite o nome do produto: ");
            vNome[i] = input.nextLine();
            System.out.println("Digite a quantidade em estoque: ");
            vQuant[i] = Integer.valueOf(input.nextLine());
            System.out.println("Digite o preço de venda: ");
            vPreco[i] = Integer.valueOf(input.nextLine());
            i++;

            System.out.println("Digite o código do produto(-1 Para cancelar): ");
            vCodigo[i] = Integer.valueOf(input.nextLine());
        }

        System.out.println("Digite o código do produto para realizar a venda(0 para cancelar): ");
        int codigo = Integer.valueOf(input.nextLine());

        while(codigo != 0){
            System.out.println("Digite a quantidade requerida: ");
            int quant = Integer.valueOf(input.nextLine());
            for (int j = 0; j < 50; j++) {
                if(codigo == vCodigo[j]){
                    produtoVendido = true;
                    if(quant <= vQuant[j]){
                        vQuant[j] = vQuant[j] - quant;
                        somaVendas = somaVendas + (vPreco[j] * quant);
                        System.out.println("Venda do " + vNome[j] + " com sucesso.");
                    }else{
                        System.out.println("Estoque insuficiente.");
                    }
                    break;
                }
            }
            if(produtoVendido = false)
                System.out.println("Produto nao cadastrado");

            System.out.println("Digite o código do produto para realizar a venda(0 para cancelar): ");
            codigo = Integer.valueOf(input.nextLine());
        }
        System.out.println("Total vendido: " + somaVendas);
        for (int j = 0; j < nProdutos; j++) {
            for (int k = 0; k < nProdutos; k++) {
                if (vQuant[j] > vQuant[k]) {
                    auxCodigo = vCodigo[j];
                    vCodigo[j] = vCodigo[k];
                    vCodigo[k] = auxCodigo;

                    auxNome = vNome[j];
                    vNome[j] = vNome[k];
                    vNome[k] = auxNome;

                    auxQuant = vQuant[j];
                    vQuant[j] = vQuant[k];
                    vQuant[k] = auxQuant;

                    auxPreco = vPreco[j];
                    vPreco[j] = vPreco[k];
                    vPreco[k] = auxPreco;
                }
            }
        }
        for (int j = 0; j < nProdutos; j++) {
            System.out.println(vCodigo[j] + " " + vNome[j] + " " + vQuant[j] + " " + vPreco[j]);
        }
    }
    
}

File: nathan/Problema0304/problema0304.py
vCodigo = []
vNome = []
vQuant = []
vPreco = []
nProdutos = 0
somaVendas = 0
produtoVendido = False

n = int(input("Digite o código do produto(-1 Para cancelar):"))
vCodigo.append(n)

while n != -1:
    nProdutos = nProdutos + 1
    x = input("Digite o nome do produto: ")
    vNome.append(x)
    n = int(input("Digite a quantidade em estoque: "))
    vQuant.append(n)
    n = int(input("Digite o preço de venda: "))
    vPreco.append(n)
    if(nProdutos == 50):
        break
    n = int(input("Digite o código do produto(-1 Para cancelar):"))
    vCodigo.append(n)

codigo = int(input("Digite o código do produto para realizar a venda(0 para cancelar):"))

while codigo != 0:
    quant = int(input("Digite a quantidade requerida: "))
    for j in range(50):
        if(vCodigo[j] == codigo):
            produtoVendido = True
            if(quant <= vQuant[j]):
                vQuant[j] = vQuant[j] - quant
                somaVendas = somaVendas + (vPreco[j] * quant)
                print("Venda do %s com sucesso" %vNome[j])
            else:
                print("Estoque insuficiente")
            break
    if(produtoVendido == False):
        print("Produto nao cadastrado")
    codigo = int(input("Digite o código do produto para realizar a venda(0 para cancelar):"))

print("Total vendido: %d " %somaVendas)

for j in range(nProdutos):
    for k in range(nProdutos):
        if (vQuant[j] > vQuant[k]):
            auxCodigo = vCodigo[j]
            vCodigo[j] = vCodigo[k]
            vCodigo[k] = auxCodigo

            auxNome = vNome[j]
            vNome[j] = vNome[k]
            vNome[k] = auxNome

            auxQuant = vQuant[j]
            vQuant[j] = vQuant[k]
            vQuant[k] = auxQuant

            auxPreco = vPreco[j]
            vPreco[j] = vPreco[k]
            vPreco[k] = auxPreco

for i in range(nProdutos):
    print("%d %s %d %d" % (vCodigo[i], vNome[i], vQuant[i], vPreco[i]))
    


